demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py

```diff
@@ -390,3 +397,51 @@ def test_load_user_metadata_bad_pack_metadata_file(repo, monkeypatch, caplog):
     pack_1_metadata.load_user_metadata("Pack1", "Pack Number 1", pack_1.path, logger)
 
     assert "Failed loading Pack Number 1 user metadata." in caplog.text
+
+
+
+
+
+@pytest.mark.parametrize("is_external, expected", [(True, ""), (False, "123")])
+def test__enhance_pack_properties__internal_and_external(mocker, is_external, expected):
+    """Tests the _enhance_pack_properties method for internal and external packs.
+    Given:
+        - Pack object.
+    When:
+        - Calling the _enhance_pack_properties method.
+    Then:
+        - Verify that the version_info is set correctly.
+        Scenario 1: When the pack is external than the version_info should be empty.
+        Scenario 2: When the pack is internal than the version_info should be set to the CI_PIPELINE_ID env variable.
+    """
+    my_instance = PackMetadata(
+    name="test",
+    display_name="",
+    description="",
+    created="",
+    legacy=False,
+    support="",
+    url="",
+    email="",
+    eulaLink="",
+    price=0,
+    hidden=False,
+    commit="",
+    downloads=0,
+    keywords=[],
+    searchRank=0,
+    excludedDependencies=[],
+    videos=[],
+    modules=[],
+) # type: ignore
+    mocker.patch(
+        "demisto_sdk.commands.content_graph.objects.pack_metadata.is_external_repository",
+        return_value=is_external,
+    )
+    os.environ["CI_PIPELINE_ID"] = "123"
```

Comment:
Please do this in a setup and teardown structure as you can influence other tests.

Reply:
better to use `monkeypatch` here instead, @RosenbergYehuda you can see examples for it all around the repo

Reply:
Fixed.